backend/spellchecker/huntojson.sh

```diff
@@ -47,24 +47,32 @@ fi
 INFILE=$1
 
 JSON=$(
-    cat $INFILE    | 
-    hunspell -a -t | 
-    grep "^&.*"    | 
-    awk $DICT '
+    cat "$INFILE" | 
+    hunspell -a -t $DICT | 
+    grep '^&.*' | 
+    sort | uniq |   
+    awk '
     BEGIN { print "{" }
     {
-        print "\t\""$2"\": [" ;
+        print "\""$2"\": [" ;
 
         split($0, split_string, ": ");
         options_number=split(split_string[2], options, ", ");
 
         for (i = 1; i <= options_number; i++) 
-            {print "\t\t\""options[i]"\","} 
+        {
+            printf "\""options[i]"\""
+            if (i < options_number)
+                { print "," }
+        }
         
-        print "\t]," }
+        print "]," }
     END { print "}" }
-    '
+    ' 
 )
 
-echo $JSON
+echo "$JSON" | 
+sed ':a;N;$!ba;s/,\n}/\n}/g' 
```

Comment:
This can probably be better solved by introducing a global variable which is empty in the beginning and "," after the first use. Use it like a prefix of the serialized array:
```
print $2 + PREFIX + [
```